### PR TITLE
[jdl] Export otherEntityField for every relationship.

### PR DESCRIPTION
**Pull request by mshima**
---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mshima** [comment]
> Why do we need it? Except for consistency.

Consistency should be enough 😄, an error should be thrown instead of ignoring those informations, otherwise.
Actually I am doing some changes that allows those relationships to be shown, but the information is just lost.

**Comment by MathieuAA** [comment]
Okay! I didn't know if there was an other reason, but I agree: consistency **is** enough.

File: test/jdl/jdl-importer.spec.js [hunk 1]
```diff
@@ -57,6 +57,7 @@ describe('JDLImporter', () => {
                     ],
                     relationships: [
                         {
+                            otherEntityField: 'region',
```
Comment (MathieuAA) [inline test/jdl/jdl-importer.spec.js:60]: Shouldn't it be `id`?

**Review by MathieuAA — COMMENTED** [review]
It's been a long time since I've had the displeasure of playing with relationships (and I'm putting it mildly ;)), but shouldn't `setRelationshipsToEntity` also be changed?

File: test/jdl/jdl-importer.spec.js (hunk 1 above)
Comment (mshima) [reply test/jdl/jdl-importer.spec.js:60]: https://github.com/jhipster/generator-jhipster/blob/da8e8bae66c9a94260305b8233129274032c6912/test/jdl/test-files/big_sample.jdl#L85
It’s correct isn’t it?

Comment (MathieuAA) [reply test/jdl/jdl-importer.spec.js:60]: `region` (the inner one) should be the displayed field. There are two possibilities here: the JDL is wrong or the `otherEntityField` is weirder than I though. Or both.
It shouldn't have been `id` but `area`, right?

Comment (MathieuAA) [reply test/jdl/jdl-importer.spec.js:60]: Tell me If I'm getting confused :laughing: I hate relationships

Comment (mshima) [reply test/jdl/jdl-importer.spec.js:60]: It’s correct, `region` is the displayed name which is `otherEntityField`.
And `area` is the `relationshipName`.

Comment (MathieuAA) [reply test/jdl/jdl-importer.spec.js:60]: Ah! Thanks :)

**Review by MathieuAA — APPROVED** [review]

**Comment by mshima** [comment]
> It's been a long time since I've had the displeasure of playing with relationships (and I'm putting it mildly ;)), but shouldn't `setRelationshipsToEntity` also be changed?

I didn’t realized `many-to-one` relationships added the other side, seems not necessary to me.
IMO it could be omitted, or I we should add `primary:true` to the main relationship so `relationship OneToMany{foo bar}` !== `relationship ManyToOne{bar foo}`
What do you think?
I will take a look at others cases.

**Comment by mshima** [comment]
> > It's been a long time since I've had the displeasure of playing with relationships (and I'm putting it mildly ;)), but shouldn't `setRelationshipsToEntity` also be changed?
> 
> I didn’t realized `many-to-one` relationships added the other side, seems not necessary to me.
> IMO it could be omitted, or I we should add `primary:true` to the main relationship so `relationship OneToMany{foo bar}` !== `relationship ManyToOne{bar foo}`
> What do you think?
> I will take a look at others cases.

Never mind, this only happens when the other side is specified.

**Pull request merged**